Documentation/Advanced.md

```diff
@@ -0,0 +1,88 @@
+# Advanced Usage #
+
+What if the endpoints you need to query don't return JSON? Perhaps you have an
+endpoint, `GET https://www.example.com/isAuthorized` that returns a `200 OK`
+with `"yes"` or `"no"` as a response body. We'll have to do a few things:
+
+- Define a custom `ResponseParser` to provide to our `Request` object, because
+the default one parses `JSON`.
+- Define a custom `Deserializer` to provide to our `APIClient`, because the 
+default one uses `NSJSONSerialization`.
+
+We'll start with the custom `Deserializer`. It will simply pass along the 
+`NSData` if it exists:
+
+```swift
+struct StringDeserializer: Deserializer {
+  func deserialize(data: NSData?) -> Result<AnyObject, SwishError> {
+    guard let data = data where data.length > 0 else {
+      return .Success(NSNull())
+    }
+
+    return .Success(data)
+  }
+}
+```
+
+Then, our custom `ResponseParser` will use `String(data:encoding:)` to
+transform the incoming `NSData` to a `String` representation:
+
+```swift
+struct StringParser: Parser {
+  typealias Representation = String
+
+  static func parse(j: AnyObject) -> Result<String, SwishError> {
+    guard let data = j as? NSData else {
+      let error = NSError.error("Bad data!")
+
+      return .Failure(.InvalidJSONResponse(error))
+    }
+
+    let error = NSError.error("Couldn't convert to string!")
```

Comment:
May or may not want to reference a Swish `NSError` `extension` in a client implementation example, could cause confusion.